src/values/AbstractObjectValue.js

```diff
@@ -233,9 +244,21 @@ export default class AbstractObjectValue extends AbstractValue {
       invariant(ob2 instanceof ObjectValue);
       let p1 = ob1.$GetPrototypeOf();
       let p2 = ob2.$GetPrototypeOf();
-      let joinedObject = Join.joinValuesAsConditional(this.$Realm, cond, p1, p2);
+      let joinedObject = Join.joinValuesAsConditional(realm, cond, p1, p2);
       invariant(joinedObject instanceof AbstractObjectValue);
       return joinedObject;
+    } else if (this.kind === "explicit conversion to object") {
+      let primitiveValue = this.args[0];
+      invariant(!Value.isTypeCompatibleWith(primitiveValue.getType(), PrimitiveValue));
+      let result = AbstractValue.createFromBuildFunction(realm, ObjectValue, [primitiveValue], ([p]) => {
+        invariant(realm.preludeGenerator !== undefined);
+        let getPrototypeOf = realm.preludeGenerator.memoizeReference("Object.getPrototypeOf");
```

Comment:
@NTillmann Should this be lazy in the builder or outside the builder? Seems like you had some thoughts on this.

Is technique guaranteed to dedupe with other residual references to these?

Such as:

```js
(function() {
  var getPrototypeOf = Object.getPrototypeOf;
  residual = function(o) {
    return getPrototypeOf(o);
  };
})();
```